python_tests/test_python_frontend.py

```diff
@@ -2258,6 +2255,95 @@ def fusion_func(fd: FusionDefinition):
         nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
         self.assertEqual(nvf_out[0], torch.gcd(inputs[0], inputs[1]))
 
+    # See https://github.com/NVIDIA/Fuser/issues/469
+    def test_repro469(self):
+        def fusion_func(fd: FusionDefinition) -> None:
+            # Note first dimension is expanded (-1, None)
+            T0 = fd.define_tensor(
+                symbolic_sizes=[-1, -1, -1, -1, -1],
+                contiguity=[None, True, True, True, True],
+                dtype=DataType.Float,
+                is_cpu=False,
+            )
+            T1 = fd.define_tensor(
+                symbolic_sizes=[-1],
+                contiguity=[True],
+                dtype=DataType.Float,
+                is_cpu=False,
+            )
+            T2 = fd.define_tensor(
+                symbolic_sizes=[-1],
+                contiguity=[True],
+                dtype=DataType.Float,
+                is_cpu=False,
+            )
+            S3 = fd.define_scalar(None, dtype=DataType.Double)
```

Comment:
```suggestion
            S3 = fd.define_scalar(dtype=DataType.Double)
```
This ^^^ works for symbolic scalar inputs.
For some reason with the optional argument, the `ValueType` for ScalarRecord becomes `bool`.

Reply:
Yeah that is strange. In this case, the test comes from the print-out of a python exception. So one temporary workaround would be to not print `None, ` in `ScalarRecord::print()` for input scalars. The heart of the issue though seems to be that `ScalarRecord::operator==()` does not behave properly when `value_ == std::nullopt`.